crates/nu-cli/src/completions/custom_completions.rs

```diff
@@ -1,19 +1,127 @@
+    list.filter_map(move |x| {
+        // Match for string values
+        if let Ok(s) = x.coerce_string() {
+            return Some(SemanticSuggestion {
+                suggestion: Suggestion {
+                    value: s,
+                    span: reedline::Span {
+                        start: span.start - offset,
+                        end: span.end - offset,
+                    },
+                    ..Suggestion::default()
+                },
+                kind: Some(SuggestionKind::Value(x.get_type())),
+            });
+        }
+
+        // Match for record values
+        if let Ok(record) = x.as_record() {
+            let mut suggestion = Suggestion {
+                value: String::from(""), // Initialize with empty string
+                span: reedline::Span {
+                    start: span.start - offset,
+                    end: span.end - offset,
+                },
+                ..Suggestion::default()
+            };
+            let mut value_type = Type::String;
+
+            // Iterate the cols looking for `value` and `description`
+            record.iter().for_each(|(key, value)| {
+                match key.as_str() {
+                    "value" => {
+                        value_type = value.get_type();
+                        // Convert the value to string
+                        if let Ok(val_str) = value.coerce_string() {
+                            // Update the suggestion value
+                            suggestion.value = val_str;
+                        }
+                    }
+                    "description" => {
+                        // Convert the value to string
+                        if let Ok(desc_str) = value.coerce_string() {
+                            // Update the suggestion value
+                            suggestion.description = Some(desc_str);
+                        }
+                    }
+                    "style" => {
+                        // Convert the value to string
+                        suggestion.style = match value {
+                            Value::String { val, .. } => Some(lookup_ansi_color_style(val)),
+                            Value::Record { .. } => Some(color_record_to_nustyle(value)),
+                            _ => None,
+                        };
+                    }
+                    "span" => {
+                        if let Value::Record { val: span, .. } = value {
+                            let start = span
+                                .get("start")
+                                .and_then(|val| val.as_int().ok())
+                                .and_then(|x| usize::try_from(x).ok());
+                            let end = span
+                                .get("end")
+                                .and_then(|val| val.as_int().ok())
+                                .and_then(|x| usize::try_from(x).ok());
+                            // TODO: error on invalid spans?
```

Comment:
We really should, but right now we can just log an error or something. I really should've done that in my PR. Not a blocker for this PR though